### PR TITLE
Global exception handling middleware

### PR DESCRIPTION
**Pull request by MishaPolishchuk**
Added global exception handling middleware

### CODE REVIEW

File: Streetcode/Streetcode.BLL/Streetcode.BLL.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="12.0.0" />
     <PackageReference Include="FluentResults" Version="3.15.1" />
     <PackageReference Include="FluentValidation" Version="11.9.0" />
+    <PackageReference Include="FluentValidation.DependencyInjectionExtensions" Version="11.9.0" />
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/Streetcode.BLL.csproj:16]: Are you sure that we need `FluentValidation.DependencyInjectionExtensions` package in `Streetcode.BLL` project ?

File: Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            await _next(context);
+        }
+        catch (ValidationException ex)
+        {
+            context.Response.StatusCode = GetStatusCode(ex);
+            context.Response.ContentType = "application/json";
+
+            var excInfoMsg = GetValidationExceptionInfoMessage(ex, context);
+            _logger.LogGlobalError(excInfoMsg);
```
Comment (MrVenik) [inline Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs:31]: Is validation exceptions should be logged ?

File: Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
+﻿// using FluentValidation;
+using Streetcode.BLL.Exceptions;
```
Comment (MrVenik) [inline Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs:2]: 'Exceptions' does not exist in the namespace 'Streetcode.BLL' ?

**Review by MrVenik — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs (hunk 2 above)
Comment (MishaPolishchuk) [reply Streetcode/Streetcode.WebApi/Middlewares/ExceptionHandlingMiddleware.cs:31]: It`s completely OK practice

File: Streetcode/Streetcode.BLL/Streetcode.BLL.csproj (hunk 1 above)
Comment (MishaPolishchuk) [reply Streetcode/Streetcode.BLL/Streetcode.BLL.csproj:16]: Seeped through. Will removed further

**Review by SvyatoS1 — APPROVED** [review]

**Review by MrVenik — APPROVED** [review]

**Pull request merged**